### ISSUE TITLE
Video trailer for Godot 2.0+

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#3877]
Ping @punto- @reduz @kubecz3k @her001 @adolson

We've discussed it a bit on IRC, a bit on Facebook, but it's better to have a place where the history is easy to access, quote and discuss further, so let's use a GH issue for that.

--------

We need a new video trailer that showcases Godot 2.0 (and as far as possible, future releases too, if we don't want to redo the trailer completely at each new release). It's a must have for the Steam store page (it currently shows older 1.1 videos), and would actually be quite cool to have on the main website and Youtube too (so probably shouldn't be made Steam-specific).

See the logs of Feb 24th for some more input and ideas: http://godot.eska.me/irc-logs/devel/2016-02-24.log (search for "trailer").

Notably, @kubecz3k proposed:
> \<kubecz3k\> Maybe something like this one: https://www.youtube.com/watch?v=UX1g5unMVD8 It's easy to edit, pleasant to watch and also easy to extend when something worthy comes out

**Resources:**

- On Facebook, @punto- asked users for footages of their games that could be used in the trailer: https://www.facebook.com/groups/godotengine/permalink/734640606672536/
  * Of particular interest IMO is @kubecz3k's https://www.youtube.com/watch?v=wYeGs3BA84w, for which the raw footage material is available at https://mega.nz/#!XBtRFKbZ!q8OK6CAGlfu78qPN0LS9FO6dZao1zSKHVzA8kMRwsCA
- GameFromScratch's showcase of Godot 2.0 (30 min): https://www.youtube.com/watch?v=av5SCu8OSS4
- @reduz's video about live editing: https://www.youtube.com/watch?v=WnpYTxCxdyI

**Tools:**

- @her001 proposed to do the editing in Blender, since he has experience with it
- Alternatively we could use Kdenlive, OpenShot or Cinelerra, but since we have a contributor willing to work on the editing in Blender already, I guess we don't need to overcomplicate it :)

--------

Please throw in ideas, link other resources and start thinking about a script, so that we can move on with this topic.

**Comment by her001** [comment]
Reasons to use Blender: Powerful, stable, supports Linux/Mac/Windows (unlike the alternative suggestions).

Reasons not to use Blender: Does not handle different fps video well (either need to use effects or an external program to work around), it isn't a dedicated video editor (shouldn't be a big deal).

If I am to be the editor, I would really prefer to use Blender.

**Comment by adolson** [comment]
I don't see any reason why Blender shouldn't be used.

One thing I would stress is to not waste time with a bunch of text or title
screen at the start. Using that UE4 video as an example, it takes 14
seconds to get to gameplay. That doesn't seem like a lot, but in this day
and age of multitasking and A.D.D., it is an eternity. I'd maybe flash a
Godot logo up there, and get right into it - 3 to 5 seconds, max.

I'd also like to hear fast-paced music, not some ethereal noise that tests
my patience.

I'm also in favor of keeping the Godot logo in a corner as a watermark.

Otherwise, I like the blurring out effect used in the UE4 video, and the
clear game and developer titles (the text might be a little large, though).

**Comment by her001** [comment]
I agree that the intro should be short. We can always have more information at the end, in case someone decides to watch that long and actually cares enough to get whatever else we might want to say (maybe have the website + logos of supported platforms or something like that).

I, personally, don't like the blur they have on there. I they might be more pleasing to me if they were faster, but I think the transition could be better as a fade or wipe.

**Comment by adolson** [comment]
I'll vote for a fade, then. I cannot stand wipes (no, not even in Star
Wars).

**Comment by punto-** [comment]
BTW this one is pretty cool too https://www.youtube.com/watch?v=3vTwtyHcLwA

**Comment by soloparatolos** [comment]
@punto I just made a video for the purpose (if you want to use it), you can edit the material as you wish. I have raw footage if needed too.
youtube: https://www.youtube.com/watch?v=4SJghbWK17A
mp4: https://www.dropbox.com/s/ijy5o0gvkx0hhwn/gravity_wars_showcase.mp4?dl=0

**Comment by akien-mga** [comment]
As resource there's also the Jetpaca video, which shows pretty cool images of https://github.com/KOBUGE-Games/jetpaca: https://www.youtube.com/watch?v=eIHAvqAqaI0

**Comment by akien-mga** [comment]
Does anybody feel like checking the various resources and put up a small script so that @her001 could start with the editing?

**Comment by her001** [comment]
@punto- requested that I make a short video with only gameplay. Here is my first pass on that:
https://www.youtube.com/watch?v=xpR3th3TKKQ&feature=youtu.be

I will add a transparent backdrop to the name-of-game text and we should find some audio to add to it.

Probably more things will change as I iterate. Any feedback is appreciated.

**Comment by punto-** [comment]
This looks awesome, I think I exaggerated a bit with the time tho now that
I see it, it goes by very fast. Can you try one with about 4 more seconds
per game? And let's look for a couple more games.

**Comment by her001** [comment]
Sounds good. Only problem: For games which are not available (to download
and record) and do not have public videos which are suitable (basically
none do), we need to ask developers to make new videos which have uncut
action for that length.

As long as we are asking, I would prefer 1080p and 60fps, and we do not
need audio.

On Wed, Mar 9, 2016 at 12:02 AM, punto- <notifications@github.com> wrote:

> This looks awesome, I think I exaggerated a bit with the time tho now that
> I see it, it goes by very fast. Can you try one with about 4 more seconds
> per game? And let's look for a couple more games.
>
> On 9 March 2016 at 01:58, Andrew Conrad <notifications@github.com> wrote:
>
> > @punto- <https://github.com/punto-> requested that I make a short video
> > with only gameplay. Here is my first pass on that:
> > https://www.youtube.com/watch?v=xpR3th3TKKQ&feature=youtu.be
> >
> > I will add a transparent backdrop to the name-of-game text and we should
> > find some audio to add to it.
> >
> > Probably more things will change as I iterate. Any feedback is
> appreciated.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/godotengine/godot/issues/3877#issuecomment-194113612
> >.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/issues/3877#issuecomment-194128078>.
>



-- 
Andrew Conrad
<aconrad103@gmail.com>

**Comment by soloparatolos** [comment]
@her001 I can try to make a couple short songs if needed or use the one in my video if you feel it could work.

**Comment by punto-** [comment]
It doesn't have to be all super exiting footage for the whole ~10 seconds,
just the thing that was happening in each clip, for 4 more seconds is fine,
especially to get an idea of the pacing I think.

**Comment by her001** [comment]
As a reference, I recorded the video myself of:
* Dolphin Island 2
* LeftWay
* Tanks of Freedom
* Dog Mendoza

Unless if I am missing a publicly accessible download link somewhere, we will need more video from the devs of:
* @kubecz3k Steam Quest
* @soloparatolos Gravity Wars (the cuts in the video you provided are too fast now)

As for other games to add, I believe the dev of Brave Bran is preparing a clip for us to use. @punto- wanted to add more games, so feel free to discuss what else we should use.

**Comment by reduz** [comment]
Awesome! If you guys can have a video for the time of GDC, we will be able
to use it to showcase it to devs there

**Comment by her001** [comment]
That was the idea ;)

On Wed, Mar 9, 2016 at 8:39 AM, Juan Linietsky <notifications@github.com>
wrote:

> Awesome! If you guys can have a video for the time of GDC, we will be able
> to use it to showcase it to devs there
>
> On Wed, Mar 9, 2016 at 11:30 AM, Andrew Conrad <notifications@github.com>
> wrote:
>
> > As a reference, I recorded the video myself of:
> >
> > - Dolphin Island 2
> > - LeftWay
> > - Tanks of Freedom
> > - Dog Mendoza
> >
> > Unless if I am missing a publicly accessible download link somewhere, we
> > will need more video from the devs of:
> >
> > - @kubecz3k <https://github.com/kubecz3k> Steam Quest
> > - @soloparatolos <https://github.com/soloparatolos> Gravity Wars (the
> > cuts in the video you provided are too fast now)
> >
> > As for other games to add, I believe the dev of Brave Bran is preparing a
> > clip for us to use. @punto- <https://github.com/punto-> wanted to add
> > more games, so feel free to discuss what else we should use.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/godotengine/godot/issues/3877#issuecomment-194319529
> >.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/issues/3877#issuecomment-194322608>.
>



-- 
Andrew Conrad
<aconrad103@gmail.com>

**Comment by vnen** [comment]
I like the new game from Kev Morelli: [Curse of the demon's sword](http://gamejolt.com/games/curse-of-the-demon-s-sword/131756). It'd be a nice add to the showcase IMO.

**Comment by soloparatolos** [comment]
@her001 Here you can grab 3 raw videos fullhd 60fps.
https://www.dropbox.com/sh/q8imhgq2vlmxl9q/AAA_N12lKe8-4iPyituoJtxka?dl=0

There is a song too in case you want to try to fit it with the content. Its loopable.

If you need more stuff let me now.

**Comment by akien-mga** [comment]
If we need more game footage, you could have a go at [Jetpaca](https://github.com/KOBUGE-Games/jetpaca). The first three levels should work fine in 2.0 (the rest may have graphical glitches or yet unsolved bugs due to moving from Godot pre-1.0 to 2.0 :)).

**Comment by adolson** [comment]
Are there any more 3D games out there?

**Comment by akien-mga** [comment]
> Are there any more 3D games out there?

Project Galatea; maybe @SaracenOne would like to make some footage for our trailer.

**Comment by her001** [comment]
For Steam Quest, that recording provided was a bit... odd. I was speeding it up a lot to make it look smooth and fit into the length. I can just make it play at nearly the same speed it was recorded at and it will be the right length now, it will just not be as smooth.

**Comment by kubecz3k** [comment]
@her001 I recorded new footage... but basically it's because the game in this setup is not running very well on my machine :( 
Take a look at those attempts: 
attempt1: https://youtu.be/63l5TBHwNX0
attemp2: https://youtu.be/wIrHdTBxeC4
raw video material:
https://mega.nz/#!7IU2jQII!wESjU8vGeUjiv2jazswJIiH5akAR-WG-x3zhcJ4bzPQ

I'm able to provide runtime with this setup for you if you are interested

**Comment by kubecz3k** [comment]
@her001 Probably tomorrow I will be able to record this on better laptop, if current resolution is not a problem (take a look at raw materials, on you tube is cropped one (not sure how rendering in kdenlive influenced the quality))  then I will record new footage :)

**Comment by her001** [comment]
Here is my latest iteration: https://www.youtube.com/watch?v=-xCxT937pLc

**Comment by adolson** [comment]
Nice! Two things:

1) maybe split the Dog title over two lines?

2) I WANT BRAVE BRAN!

**Comment by her001** [comment]
I can't believe I didn't think to try wrapping that! Tomorrow I'll try it out.

**Comment by akien-mga** [comment]
Looks very nice! Same comment as @adolson for the wrapping, and indeed Brave Bran has some speed issues :)

**Comment by zoiba** [comment]
Hello, I've got an email from Dana who asks for a clip from the game we are making with Godot. Here a quick capture: https://youtu.be/GFmt7tjE1Jk and the mp4 file from here http://ffxrunner2.ffx.it/download/ffxrunner2_20160310.mp4
don't know if the video quality is enough, please let me know, we will try a better capture in the case.

**Comment by kubecz3k** [comment]
@her001 I recorded new footage, it should be more fluent since I got a lot more fps this time.
Raw materials: https://mega.nz/#!7IFQGCBL!sVPD1kNV6ZYYw_S3DWrrQdcjeY96Uu8-RCD2Z5pUu1I

**Comment by her001** [comment]
Here is my latest iteration. I made a little song specifically to go with it also.
https://www.youtube.com/watch?v=rdQwCj-_Rqc

**Comment by soloparatolos** [comment]
The video is ok, but I feel like the song is a bit slow/dark for the purpose. Just my opinion, music stuff is quite a subjective thing.

**Comment by akien-mga** [comment]
I really like the video, awesome job! As for the song, it's OK but I'm personally not too fond of typical soundfont sounds like the strings, choirs and horns you're using; not that it sounds bad, but it has IMO a kind of amateur vibe to it. I'm pretty sure we could find well-suited CC BY or BY-SA pieces that would fit the trailer too.

**Comment by akien-mga** [comment]
Also, @reduz has some pretty cool chiptune pieces, if we find one that fits the trailer I'm pretty sure he would happily license it for using with the video :) https://soundcloud.com/reduz

**Comment by akien-mga** [comment]
Throwing some links from OGA:
- http://opengameart.org/content/agent-6 Somewhat Sci-Fi chiptune, it's quite long so would have to be edited to fit our short trailer
- http://opengameart.org/content/more-time-2
- http://opengameart.org/content/orchestra-loop-0 Could sound pretty epic, but it lacks some Woodkid-like drums to bring even more power after a couple loops :)

Alexandr Zhelanov has many epic sounding tracks: http://opengameart.org/users/alexandr-zhelanov e.g.:
- http://opengameart.org/content/low-fi <== This one might be pretty fitting actually
- http://opengameart.org/content/rise-of-spirit With some editing, could be nice
- http://opengameart.org/content/betrayed Pretty epic, but would need a similarly epic trailer fitted to it, not a showcase of various games
- http://opengameart.org/content/open-the-gate A bit too "grandiose" maybe, sounds like Vangelis :p
- http://opengameart.org/content/we-will-meet-again
- http://opengameart.org/content/its-our-battle
- http://opengameart.org/content/battle-music
- http://opengameart.org/content/intro-0

Ok I'll stop there for Alexandr's musics, there are 5 pages more that I haven't listened to hehe.

Not all of the above are suited for our current showcase, but they might be useful for the longer version of the trailer maybe.

**Comment by akien-mga** [comment]
Of course the advantage of @her001's music is that it has been edited to fit the various footages, so that's also something to consider :)

**Comment by her001** [comment]
Hehe. The music is certainly is all about taste. I only made it because no one provided anything that wasn't already in a source video. I am not sure I even like my song that much, but I did put some effort into making it match the video.

@soloparatolos: It is funny, I thought it might have been too fast or happy.

@akien-mga: I could, of course, replace instruments (most are, in fact, soundfonts). I didn't go with anything chiptune  as I did not want to give the impression that Godot is for retro games. I know that chiptune does not mean retro, but I think the song can be as important as the video when it comes to controlling expectations.

I am only explaining my reasoning here. If there are enough people that think it should be modified or replaced, I will gladly do so.

**Comment by reduz** [comment]
My only feedback about it would be to:

1) Reverse the order, I think we should should show the visually stronger
titles first, like Brave Bran, Dog Mendonca, SteamQuest, Dolphin Island
2) I would include just a little more footage from Dog Mendonca
3) I always use music from this guy in the Godot videos, who has kindly
allowed it (and he rules) - https://soundcloud.com/madbr

**Comment by SaracenOne** [comment]
@akien-mga 
Hello, I just saw the interest in potentially using some video footage in my 3D project as part of the trailer. Personally, I would love to get some footage in there, but unfortunately at this point, the current state it's in is a bit messy because I'm heavily experimenting with a lot of things, meaning its full of glitches and massive slowdowns, so I'm not really able to produce any new footage at this exact moment. One thing I can offer is some of the preexisting footage to see if you can do anything with it.

At the same, a lot of things being shown off are not yet avaliable in the vanilla engine yet, so might be slightly misrepresentitive. Things like IK and procedural skyboxes are stuff I'm currently converting to work as C++ modules. Likewise, the improved facial morph system, I would actually be attempting to get merged right now, but I'm trying to develop a new intermediatary file format to replace the slightly hacky workflow currently required to get them in the game, and I'm also working on a new minigame right now which would do a very good job of showing off this feature, but it's just not ready yet.

Either way, if there I've already produced which you would really like to use, please let me know. Even if I'm not able to get anything in this trailer in time, I would love to be involved in the future. I kind of hoped that this project could serve a duel purpose of being representitive of the higher-end of this engine's ever evolving 3D capabilities at some point, so getting it in an official trailer would be great (:

**Comment by her001** [comment]
Okay, I've made a few alternative videos.

https://www.youtube.com/watch?v=jwDtbc8WyW8
This replaces the audio track with [Rainbow Power](https://soundcloud.com/madbr/rainbow-power) by madbr. It conveniently fits the length of the video.

https://www.youtube.com/watch?v=SEMIpZFYyKA
This also uses Rainbow Power, but has the video clips rearranged to match @reduz's request to have the visually impressive games in front. I do not like this one very much. I think it messes with the pacing in a bad way.

https://www.youtube.com/watch?v=BRR58n_ONzQ
This is has the original order of the clips and replaces the music with the same as what I had, but new instruments. Could have balance issues between the instruments, and if you didn't like the song itself you probably still won't like it.

For comparison, here is the latest one I posted before: https://www.youtube.com/watch?v=DYGzAyRmle8

**Comment by adolson** [comment]
As far as clip order, after watching both, I prefer the original. I think
Steam Quest looks strong at the start, and it's good to end on a high note.
LeftWay is very stylized, even more so than the pixelart games, so it is
good in the middle of the pack, rather than as an opener or closer.

As for the music, I do not like the new instrument version. Everything
sounds out of tune and not as good as the original. Between the two songs,
I don't really have a preference. Neither is my style, but I may be in the
minority of people who would rather hear distorted guitars, chunky bass,
and double-kick.

**Comment by her001** [comment]
From my little bit of asking people not participating in this issue I have:
2 people for [RC3](https://www.youtube.com/watch?v=jwDtbc8WyW8)
1 person for [RC4](https://www.youtube.com/watch?v=SEMIpZFYyKA)
2 for [RC5](https://www.youtube.com/watch?v=BRR58n_ONzQ)
1 person for anything but RC3, including [RC2](https://www.youtube.com/watch?v=DYGzAyRmle8)

The one for RC4 was freeman on #godotengine-dev, who is interested in making a song specifically for the edit in that video.

**Comment by akien-mga** [comment]
I prefer the original clip order which is indeed more balanced. Showing off the best-looking games first to end some less impressive - albeit quite good too - games kind of unbalances it all; and the video is still short enough that people won't say "boring" after 15 s and miss Brave Bran at the end IMO.

For the music, I think Rainbow Power is quite nice, to my ears it sounds better than RC2 or RC5 (this of course purely a matter of taste, but beyond the soundfont instruments that I'm not fond of, I find that parts of it sound dissonant, as if all tracks were not playing on the same key as the rest - especially the 2nd track to start, not sure what kind of instrument it is).

**Comment by zoiba** [comment]
we got some better footage: https://www.youtube.com/watch?v=I7C4TRoWQ_8
and here the movie file: http://ffxrunner2.ffx.it/download/ffxrunner2_20160311.mp4

**Comment by freemanfromgodotengine** [comment]
Like Her001 already mentioned, I really thought Godot's community deserves better music in new trailer, so I decided to spend this night making one myself.

You may check trailer with new music here.
https://youtu.be/CXN3cGnbE0M
The audio file for Hero_01 if you decide that you like the music, can be downloaded from my repository:
https://github.com/freemanfromgodotengine/my_repository/blob/master/Godot2_0_video_trailer/godot%20trailer%20music%20proposal%20by%20freeman.wav

**Comment by akien-mga** [comment]
@freemanfromgodotengine I really like the music you proposed!

**Comment by adolson** [comment]
I like the music @freemanfromgodotengine ! But I hate that clip order... Although maybe it would be fine if a clip from @zoiba is added at the end... But I still feel the original order is better. And I'd add a clip from @zoiba anyhow, since that is honestly the best-looking 3D game I've seen in Godot so far, and the one that I want to play the most.

**Comment by her001** [comment]
It is amazing how different everyone's opinion is.

@freemanfromgodotengine, I really like your song, but I do not like that order of the videos still.

@zoiba, if it is decided that your video is to be included, there probably won't be time for it to be in the showcase by GDC. Also, if you have raw, uncut video, that would be preferred.

**Comment by her001** [comment]
Here is the alternate order of the video with @freemanfromgodotengine's song, so that you can view it at a higher quality (when it is done processing). https://www.youtube.com/watch?v=Vx9fPei12i0

**Comment by akien-mga** [comment]
I would propose to use the video by @freemanfromgodotengine for the GDC showcase for the time being, as it's quite good already IMO.

Then we should keep working on it in parallel to make a showcase that everyone will like, to put on our website, on the Steam store page, on an official Godot YouTube channel, etc. To do so, I think we need to:
- Add @zoiba's game footage to the showcase
- Define the "final" ordering of the footages
- Have @freemanfromgodotengine rework his song to the ordering we'd have settled upon (as he made each part fitting to a specific game footage, so I think just changing the order of the videos and keeping the same song won't work as well as intended).

**Comment by punto-** [comment]
Update from the field: the gameplay trailer works great, but I find that
when I'm showing it to someone, I need about 4-5 seconds more on each game
to be able to describe the game and whatever brief story I need to say
about the developer (yes i know I originally asked for 30 seconds and now
it's 1 minute and still not enough, I was wrong :p ). Other than that, it's
great!

**Comment by her001** [comment]
@punto- Glad to hear it is working alright.  It will be quite difficult for me to make a new version right now, unfortunately, but I can try if you would like.

* I too am away from home for a few days. I can copy all the files to my laptop, but downloading and working on them will be a bit slow.
* We might need new footage for some of the games (but maybe not, I'd have to go through the sources).
* We will definitely need new music.

**Comment by akien-mga** [comment]
Nice to get some feedback from the field :D We should however keep in mind that the typical use case of this video will not be as a presentation support for Godot developers explaining how cool their engine and community are, but more as a short video that can be used on Steam, on our website, or embedded in press reviews of Godot. So I think the current format is fine as is; @punto- need for more footage of each game while giving insider tips will probably not happen that often :)

For this purpose, we should probably more work on the second trailer we had considered, that shows game footages + editor footages of those games, thus showing the editor in action and how user-friendly and powerful it is.

**Comment by punto-** [comment]
I would say if we can have something to use during Thursday and Friday, it's worth the effort for us, but if you're out of town don't worry about it, enjoy your trip :) I'll speak faster :pFor the music, I noticed how well it matches the game changes, but in this case it's not that important because we're showing it in places where there's usually noise and we're speaking over it, so it's not a big deal of it's out of sync, we can fix that later to upload to Steam. As Remi says tho, this pacing issue night not translate to that version, where you're playing it with headphones on the Steam page, so this version might be already perfect for that.                                                                                                                                                                                                   Sent from my BlackBerry 10 smartphone.                                                                                                                                                                                                                From: Andrew ConradSent: Tuesday, March 15, 2016 7:45 AMTo: godotengine/godotReply To: godotengine/godotCc: punto-Subject: Re: [godot] Video trailer for Godot 2.0+ (#3877)@punto- Glad to hear it is working alright.  It will be quite difficult for me to make a new version right now, unfortunately, but I can try if you would like.


I too am away from home for a few days. I can copy all the files to my laptop, but downloading and working on them will be a bit slow.
We might need new footage for some of the games (but maybe not, I'd have to go through the sources).
We will definitely need new music.


—You are receiving this because you were mentioned.Reply to this email directly or view it on GitHub:https://github.com/godotengine/godot/issues/3877#issuecomment-196848757

**Comment by freemanfromgodotengine** [comment]
I could adjust my soundtrack for the new version of the trailer, if you want me to, but I would need a final video file and some time to be able to work with it.
In the worst case scenario I would need a precise timing of things happening in the new version of the trailer, things like when the next scene with new game is shown, when the logo enters the screen, if there are any visual accents and when etc..
And frankly, if some version of this trailer is going to be uploaded on Steam (I didn't know about that), I think we should take our time and try to make it better.
I will gladly cooperate with **her001** to bring music and visuals even closer together.

@punto-  

> "I find that when I'm showing it to someone, I need about 4-5 seconds more on each game to be able to describe"

I think the present pace of the changes of scenes works pretty good for the purpose of the video on YT or Steam.
If there is a need for longer version that would be used for live presentations by devs, I think we could create another version with that in mind.

**Comment by akien-mga** [comment]
Hi guys,

I was contacted by a freelance journalist that wants to write a small article on Godot and have some pictures (for printing in a magazine). This brings the question of the licensing of the content that we (as Godot developers) got from community members and use to showcase the engine features. And that concerns the video trailer too, as we should license it in a way that third parties know what they are allowed to make with it.

In my opinion, the ideal way would be to license the video trailer under [CC BY-SA 4.0 International](https://creativecommons.org/licenses/by-sa/4.0/) with copyright for "Juan Linietsky, Ariel Manzur and the Godot community", as we do for the docs and the website (since there is no legal "Godot Foundation" entity to attribute the copyright to, so I *guess* it still has to be nominal).

However, I fear that to apply such a license we'd need to have all sub-footages and the music under a compatible license, so we'd have to assess for each of the footages and the music if their authors are fine with that:

- Music track - @freemanfromgodotengine 
- Brave Bran footage - @ndee85
- Dog Mendonça footage - @punto- (might be harder to get if it needs to be approved by the publisher I guess?)
- Steam Quest footage - @kubecz3k 
- Dolphin Island 2 footage - would have to contact the devs directly
- Gravity Wars - @soloparatolos 
- Tanks of Freedom - @w84death 
- LeftWay - would have to contact the dev directly

I'm not asking you guys for a formal licensing right now, just throwing out some ideas so that we can find what should be done; maybe some of you have experience with such things already.

Another solution would be to keep the trailer as all rights reserved, with a formal permission grant from the above mentioned authors to use their work in the trailer + to allow the Godot developers to grant distribution permissions themselves (so that online media can show our video too, etc., without having to ask each copyright holder each time if we can diffuse the footage from their game).

All in all I'm a bit clueless, hopefully some of you have experience with that already, if not I'll probably try to contact Conservancy to get some legal guidance.

**Comment by w84death** [comment]
We (P1X/Tanks of Freedom) are happy with the CC-BY licence. I've already changed that in the YouTube.

Our video is made primary for the press and its purpose is to be shown as much as possible :)

**Comment by her001** [comment]
As editor, I am fine with any CC license for the finished product.

It should be noted that I recorded the footage for Dog Mendonça, Dolphin
Island 2, Tanks of Freedom, and LeftWay. If the devs have a blanket
statement about using recorded footage that would allow us to use it, we
don't *need* to ask (though it wouldn't hurt to ask anyway).

On Thu, Mar 17, 2016 at 5:54 AM, Krzysztof Jankowski <
notifications@github.com> wrote:

> We (P1X/Tanks of Freedom) are happy with the CC-BY licence. I've already
> changed that in the YouTube.
>
> Our video is made primary for the press and its purpose is to be shown as
> much as possible :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/issues/3877#issuecomment-197818111>
>



-- 
Andrew Conrad
<aconrad103@gmail.com>

**Comment by ndee85** [comment]
As the creator of the video footage for Brave Bran and creator of Brave Bran itself I am also absolutely fine the CC license for the video.

**Comment by soloparatolos** [comment]
Im the one behind Gravity Wars and Im fine with the CC license too.

**Comment by akien-mga** [comment]
I've done a bit of thinking about this topic, and I think getting the whole video licensed as CC By-SA would be difficult; I could not find relevant legal guidance about this so far, but I believe licensing the complete video under a creative commons license would imply that all gameplay videos should bear the same license. And gameplay videos are kind of a difficult legal area that I don't know much.

Simple reasoning: we distribute the video as CC By-SA, someone takes a screenshot of the Dog Mendonça gameplay, one could argue that's it also a CC By-SA derivative work, and then one could cut out the Pizza Boy sprite in GIMP and reuse it in another game as CC By-SA. That sounds ethically crazy, but up to now I haven't found legal evidence that this scenario would not be possible; some good information of what licensing videos implies on the content shown in the video would be appreciated :)

Here's what the [Youtube policy](https://support.google.com/youtube/answer/2797468?hl=en) for CC By content says:
> Please understand that you may only mark your uploaded video with a Creative Commons license if it consists entirely of content licensable by you under the CC BY license.

All in all, I feel this is more hassle than gain, so maybe it's not the way to go.

------------

So, here's what I'd propose for now:

- **Keep the full video non-libre/all rights reserved**, which means that only we (Godot Engine) are allowed to make derivative works of it, etc. That means that on Youtube we'd keep the "Standard Youtube license"
- Mention in the description (on Youtube and other platforms) the copyrights information for each subfootages (e.g. something like "The art and gameplay footages used in this video remain the property of their respective copyright holders, and are licensed for usage to Godot Engine in promotion material." + a list of game names and respective authors).

As much as I love free culture and Creative Commons, trying to use it for a collection of gameplay footages from copyrighted (and a priori not freely-licensed) games is too much headache for us to bear with legal guidance :)

*In conclusion*, what we'd ask of all copyright holders to avoid having to bother too much with this each time we need to reuse the material, would be to grant a **usage permission** for the footages to "Juan Linietsky, Ariel Manzur and the Godot contributors" (i.e. the "legal" Godot as mentioned before). That's something that most of you basically already did by accepting that the video would be CC licensed when I proposed it, so we probably don't need to be overly bureaucratic and to get this in written form, IMO the above discussion is enough. I'll just need to get a confirmation by @punto- that we're allowed to use the Dog Mendonça footage, and get in touch with the LeftWay and Dolphin Island 2 devs.

Sorry if this all sounds overly paranoid and boring; but from my experience those are not topics that should be taken lightly (not that I fear any kind of copyright infringement notice from any of you hehe, but we need to make sure we know what we're doing from the start).

**Comment by vnen** [comment]
It could be licensed as CC By-ND (using a "No Derivative" clause), so anyone can distribute the video, but nobody is allowed to re-edit and much less to use the shown sprites in other games.

**Comment by reduz** [comment]
Video was a success with anyone I showed it to btw. I'll write more when
gdc over and also share views with punto

**Comment by freemanfromgodotengine** [comment]
Like the later post by @akien-mga , I would also prefer if our work was freely distributable but not modified by others.
As an author of the music for the trailer I would gladly transfer my right to Godot Foundation or whoever is going to in charge and let them decide who to give permission to adjust, shorten or change the work as required. In case that would create mail flood from TV stations with questions about using fragments etc., Godot always may prepare the addendum to the license, that showing shortened version in the context of Godot review or Godot related video footage is permitted.

@vnen  already proposed and I 2nd his opinion, that CC By-ND is probably what we are looking for.

As of 'CC BY-SA 4.0 International', as much as wanted to share the music with trailer creators in the goal to make even better trailer, some kid recording his girlfriend singing "Godot sucks" on top of my music and also putting my name/nick there does not makes me thrilled, though.

@reduz Good to hear that. Looking forward for the coverage from GDC. Don't forget to use some photos and video.

**Comment by ndee85** [comment]
@reduz nice to hear. Would be really cool to have a gdc report once you are back! :+1:

**Comment by Mavhod** [comment]
Hi, I'm LeftWay game developer.

Thanks for add my game in Godot engine showcase :D, I'm glad for you use it.

And sorry for late answer, and thank you her001 for made this video.

**Comment by janmarcano** [comment]
Hi!
Dolphin Island's developer here. I'm alright with the License being proposed and with footage of the game being used.

Also, @freemanfromgodotengine: Love the music! (Specially the bars that go along the Di2 footage :P)

**Comment by freemanfromgodotengine** [comment]
Thx @janmarcano . That fragment of the music was inspired by the footage.

**Comment by akien-mga** [comment]
So, I finally created an official Youtube channel for Godot and uploaded the trailer there: https://www.youtube.com/channel/UCKIDvfZD1ZhY4_hhbotf7wA

I used CC BY-ND as license as proposed by @vnen, it's a good compromise IMO.

**Comment by her001** [comment]
Awesome. Things that still need to be done (I think): Put this on Steam,
decide on a way to make public the source files (recordings and blender
files), make an updated features video.

On Sun, Apr 3, 2016 at 2:59 PM, Rémi Verschelde <notifications@github.com>
wrote:

> So, I finally created an official Youtube channel for Godot and uploaded
> the trailer there:
> https://www.youtube.com/channel/UCKIDvfZD1ZhY4_hhbotf7wA
>
> I used CC BY-ND as license as proposed by @vnen <https://github.com/vnen>,
> it's a good compromise IMO.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/issues/3877#issuecomment-205045183>
>



-- 
Andrew Conrad
<aconrad103@gmail.com>

**Comment by akien-mga** [comment]
Thanks again everyone for the awesome work done here! I think we can consider this issue fixed, and it is now superseded by #5917 which aims at creating a new video trailer for Godot 2.1, which would be more focused on showcase the editor itself :)
